firestore/incidents.py

```
@@ -38,29 +40,72 @@ class Meta:
 
 
 class Incident(BaseReport):
-    incident_time = mdl.DateTime(required=True)
-    incident_location = mdl.TextField()
-    abstract = mdl.TextField(required=True)
-    abstract_translate = mdl.MapField(required=True)
+    abstract = mdl.TextField(required=False)
+    abstract_translate = mdl.MapField(required=False)
     url = mdl.TextField(required=False)
-    incident_source = mdl.TextField(required=True)
+    incident_source = mdl.TextField(required=False)
     created_by = mdl.TextField(required=False)
     title = mdl.TextField(required=False)
     title_translate = mdl.MapField(required=False)
     parent_doc = mdl.TextField(column_name="parent")
 
 
+VALID_SELF_REPORT_STATUSES = {"all", "approved", "rejected", "new"}
+VALID_TYPE_STATUSES = {"news", "self_report", "both"}
+
 @cached(cache=INCIDENT_CACHE)
-def queryIncidents(start: datetime, end: datetime, state=""):
+def queryIncidents(start: datetime, end: datetime, state="", type="", self_report_status="", start_row="", page_size=""):
+    # Convert start_row and page_size to integers
+    # TODO: implement the pagination based on Firestore (https://firebase.google.com/docs/firestore/query-data/query-cursors)
+    try:
+        # Validate type and self_report_status
+        if self_report_status not in VALID_SELF_REPORT_STATUSES:
+            return {"error": f"Invalid self_report_status: {self_report_status}. Allowed values are {VALID_SELF_REPORT_STATUSES}"}
+        if type not in VALID_TYPE_STATUSES:
+            return {"error": f"Invalid type: {type}. Allowed values are {VALID_TYPE_STATUSES}"}
+        start_row = int(start_row) if str(start_row).isdigit() and int(start_row) >= 0 else 0
```

Why do we still need to convert `start_row` and `page_size` to `string` given they are `string` type?

E.g. if `start_row` is not a `string`, should we raise an error here as it is not expected? If it is a `string` type, we can simply use `start_row.isdigit()`.

Another question is, you only checked the lower-bound of `start_row` and `page_size`, don't you need to check the upper-bounds? E.g. `start_row` shouldn't be greater than the length of `incidents` right?